src/controllers/auth.controller.js

```diff
@@ -0,0 +1,189 @@
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+};
+
+const registration = async (req, res, next) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    name: validateName(name),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.name || errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPass = await bcrypt.hash(password, 5);
+
+  await userService.register(name, email, hashedPass);
+
+  res.send({ message: 'OK' });
+};
+
+const changePassword = async (req, res) => {
+  const { password, confirmation } = req.body;
+  const { activationToken } = req.params;
+
+  if (password !== confirmation) {
+    throw ApiError.badRequest('Values should match');
+  }
+
+  if (validatePassword(password)) {
+    throw ApiError.badRequest(validatePassword(password));
+  }
+
+  const user = await userService.changePassword(password, activationToken);
+
+  await user.save();
+
+  generateTokens(res, user);
+};
+
+const reset = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw ApiError.badRequest('Email is not provided');
+  }
+
+  await userService.resetPassword(email);
+  res.sendStatus(204);
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    return res.sendStatus(404);
+  }
+
+  user.activationToken = null;
+  await user.save();
+
+  await generateTokens(res, user);
+};
+
+const generateTokens = async (res, user) => {
+  const normalizedUser = userService.normalize(user);
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshAccessToken = jwtService.signRefresh(normalizedUser);
+
+  await tokenService.save(normalizedUser.id, refreshAccessToken);
+
+  res.cookie('refreshToken', refreshAccessToken, {
+    HttpOnly: true,
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+  });
+
+  res.send({
+    user: normalizedUser,
+    accessToken,
+  });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  if (user.activationToken) {
+    throw ApiError.badRequest('Activate your account. Check you email');
+  }
+
+  await generateTokens(res, user);
+};
+
+const refresh = async (req, res) => {
+  const { refreshToken } = req.cookies;
+
+  const userData = jwtService.verifyTokenRefresh(refreshToken);
+
+  if (!userData) {
+    throw ApiError.unauthorized();
+  }
+
+  const token = await tokenService.getByToken(refreshToken);
+
+  if (!token) {
+    throw ApiError.Unauthorized();
```

Comment:
There is a typo in `ApiError.Unauthorized()`. The method name should be `unauthorized()` to match the casing used in other parts of the code.